src/query/bool.rs

```
@@ -0,0 +1,89 @@
+use super::{CreateQuery, Result, TermQueries};
+
+use tantivy::query::{BooleanQuery, Occur, Query};
+use tantivy::schema::Schema;
+
+#[derive(Deserialize, Debug, PartialEq)]
+pub struct BoolQuery {
+    #[serde(default = "Vec::new")]
+    must: Vec<TermQueries>,
+    #[serde(default = "Vec::new")]
+    filter: Vec<TermQueries>,
+    #[serde(default = "Vec::new")]
+    must_not: Vec<TermQueries>,
+    #[serde(default = "Vec::new")]
+    should: Vec<TermQueries>,
+    minimum_should_match: Option<u64>,
+    boost: Option<f64>,
+}
+
+impl CreateQuery for BoolQuery {
+    fn create_query(self, schema: &Schema) -> Result<Box<Query>> {
+        let mut all_queries: Vec<(Occur, Box<Query>)> = Vec::new();
+        all_queries.append(&mut parse_queries(schema, Occur::Must, &self.must)?);
+        all_queries.append(&mut parse_queries(schema, Occur::MustNot, &self.must_not)?);
+        all_queries.append(&mut parse_queries(schema, Occur::Should, &self.should)?);
+        Ok(Box::new(BooleanQuery::from(all_queries)))
+    }
+}
+
+fn parse_queries(schema: &Schema, occur: Occur, queries: &[TermQueries]) -> Result<Vec<(Occur, Box<Query>)>> {
+    let mut results = Vec::with_capacity(queries.len());
+    for q in queries.into_iter() {
+        match q {
+            TermQueries::Fuzzy(f) => {
+                let fuzzy_query = f.clone().create_query(&schema)?;
+                results.push((occur, fuzzy_query));
+            }
+            TermQueries::Exact(q) => {
+                let term_query = q.clone().create_query(&schema)?;
+                results.push((occur, term_query));
+            }
+            TermQueries::Range(r) => {
+                let range_query = r.clone().create_query(&schema)?;
+                results.push((occur, range_query));
+            }
+            _ => unimplemented!(),
```

should probably return an error?